Cleanup setup.py/tox/etc.

aragilar opened the pull request.

aragilar committed:
Update tox to use pyproject.toml

aragilar committed:
Add python 3 support to benchmarks

aragilar committed:
Add more checkers to tox

Add flake8, check-manifest and checkreadme. Add these to github actions,
and add pypy3 to CI.

aragilar committed:
Use src dir to avoid issues with imports

Previously the tests were running against the non-installed version.

pep8speaks commented:
Hello @aragilar! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:













































* In the file [`docs/conf.py`](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/docs/conf.py):

> [Line 55:1](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/docs/conf.py#L55): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file

* In the file [`setup.py`](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py):

> [Line 146:80](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py#L146): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 163:17](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py#L163): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 163:19](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py#L163): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 164:20](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py#L164): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 164:22](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/setup.py#L164): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals





























* In the file [`tests/test_compile.py`](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/tests/test_compile.py):

> [Line 5:1](https://github.com/aragilar/logbook/blob/21d36aefafb3ec6a915caec44bb7c66a1b434260/tests/test_compile.py#L5): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1



##### Comment last updated at 2022-06-03 12:52:42 UTC

aragilar committed:
Add riemann_client extra

aragilar committed:
Add compile check to replace custom check

This simplifies the tox config.

aragilar committed:
Add windows to CI

aragilar committed:
Pass through ENABLE_LOGBOOK_NTEVENTLOG_TESTS

This should mean across all platforms we're skipping no tests.

aragilar committed:
Update MANIFEST.in based on CI

Drop Vagrantfile as it is significantly out of date, and docker would
likely be a better tool to use.
